### PR TITLE
DRIVERS-2789 Convert Benchmarks Spec to Markdown

### PR DESCRIPTION
**Pull request by blink1073**
Please complete the following before merging:

- [x] Update changelog.
- [ ] Make sure there are generated JSON files from the YAML test files.
- [ ] Test changes in at least one language driver.
- [ ] Test these changes against all server versions and topologies (including standalone, replica set, sharded
  clusters, and serverless).

### CODE REVIEW

File: source/benchmarking/benchmarking.md [hunk 1]
```diff
@@ -0,0 +1,689 @@
...
+
+### Datasets
+
+TBD: generated datasets should be park in S3 or somewhere for retrieval by URL
+
+## FAQ
+
+Question?
+
+Answer.
```
Comment (alcaeus) [inline source/benchmarking/benchmarking.md:655]: I feel like this is a good opportunity to remove this empty section.

File: source/benchmarking/benchmarking.md [hunk 2]
```diff
@@ -0,0 +1,689 @@
...
+of the following types: string, Int32, Int64, Double, Boolean. (121 total key/value pairs) Keys will be random ASCII
+strings of length 8. String data will be random ASCII strings of length 80.
+
+Dataset size: For score purposes, the dataset size for a task is the size of the single-document source file (7531
+bytes) times 10,000 operations, which equals 75,310,000 bytes or 75.31 MB.
+
+Phases:
+
+|             |                                                                                                                                                                                              |
+| ----------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
```
Comment (alcaeus) [inline source/benchmarking/benchmarking.md:169]: This and all other tables here have an empty row at the top.

**Review by alcaeus — COMMENTED** [review]
Couple of minor comments, but looking good!

File: source/benchmarking/benchmarking.md (hunk 2 above)
Comment (blink1073) [reply source/benchmarking/benchmarking.md:169]: Github Flavored Markdown doesn't support tables without headers

File: source/benchmarking/benchmarking.md (hunk 1 above)
Comment (blink1073) [reply source/benchmarking/benchmarking.md:655]: Done

File: source/benchmarking/benchmarking.md (hunk 2 above)
Comment (blink1073) [reply source/benchmarking/benchmarking.md:169]: I'll make up some headers.  :smile:

**Review by alcaeus — APPROVED** [review]

File: source/benchmarking/benchmarking.md (hunk 2 above)
Comment (alcaeus) [reply source/benchmarking/benchmarking.md]: Oh, I wasn't aware of that. Thanks for making up some headers!

**Pull request merged**